examples/cc2538-common/test-uart.c

```diff
@@ -0,0 +1,113 @@
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file is part of the Contiki operating system.
+ *
+ */
+/**
+ * \addtogroup cc2538-examples
+ * @{
+ *
+ * \defgroup cc2538-test-uart Test the CC2538 UART
+ *
+ * Demonstrates the use of the CC2538 UART
+ *
+ * @{
+ *
+ * \file
+ *         A quick program for testing the UART1 on the cc2538 based
+ * platform which can be used to interface with sensor with UART
+ * interface and to communicate with other development boards
+ * \author
+ *         Akshay P M <akshaypm@ece.iisc.ernet.in>
+ */
+/*---------------------------------------------------------------------------*/
+#include "contiki.h"
+#include "sys/etimer.h"
+#include "sys/rtimer.h"
+#include "dev/leds.h"
+#include "dev/uart.h"
+#include "dev/serial-line.h"
+#include "dev/slip.h"
+#include <stdio.h>
+#include <stdint.h>
+/*---------------------------------------------------------------------------*/
+static struct etimer et;
+/*---------------------------------------------------------------------------*/
+PROCESS(cc2538_uart_demo_process, "cc2538 uart demo");
+AUTOSTART_PROCESSES(&cc2538_uart_demo_process);
+/*---------------------------------------------------------------------------*/
+unsigned int
+uart1_send_bytes(const unsigned char *s, unsigned int len)
+{
+  unsigned int i = 0;
+
+  while(s && *s != 0) {
+    if(i >= len) {
+      break;
+    }
+    uart_write_byte(1, *s++);
+    i++;
+  }
+  return i;
+}
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(cc2538_uart_demo_process, ev, data)
+{
+  uint16_t rxdata = 0;
+  static uint16_t count=25;
+  char string[count];
+  extern process_event_t serial_line_event_message;
```

Comment:
You should not declare `serial_line_event_message`, because it is declared by `"dev/serial-line.h"`.

`count` holds the number of each line that is sent out through the UART; therefore, it should be initialized to zero.
The `string` array should be given a constant size; the size should be big enough to hold the longest string that the `sprintf()` call might create.  Its format string has 11 bytes that don't change.  `count` is 16 bits; therefore, the biggest number is 65535.  So, 11 + 5 is 16; let's extend the array a little bit, and use 18.
The above three lines should be changed into these two lines:
```c
char string[18];
static uint16_t count = 0;
```

(The rest of the code looks good.)